### PR TITLE
feat: Add docs for advanced datascrubbing

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 1]
```diff
@@ -0,0 +1,425 @@
+---
+title: 'Advanced Data Scrubbing (Preview)'
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Do we want to replace "(Preview)" here with "Beta" as we discuss it as a beta option in the text that follows?

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+sidebar_order: 1
+keywords: ["pii", "gdpr", "personally identifiable data", "compliance"]
+---
+
+In addition to using [`beforeSend`]({% link _documentation/data-management/sensitive-data.md %}#custom-event-processing-in-the-sdk) in your SDK or our [server-side data scrubbing features]({% link _documentation/data-management/sensitive-data.md %}#server-side-scrubbing) to redact sensitive data, we are currently beta-testing ways to give you more granular control over what exactly our server-side data scrubbing does to your events. Additional functionality includes:
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: remove hyphen in "beta-testing"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+
+* Ability to define custom regular expressions to match on sensitive data
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest rephrase to "Define custom regular expressions..."

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+* More fine-grained tuning on which parts of an event to scrub
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "Detailed tuning of the parts of an event to scrub"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+* Partial removal or hashing of sensitive data instead of just deletion
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggestion - remove "just"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+
+## Overview
+
+**Advanced Data Scrubbing is currently only available if your organization is an Early Adopter.** To become one, go to your organization settings and enable the "Early Adopter" option to get access to new features before they're fully released. This option can be just as easily disabled as it can be enabled.
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "Advanced Data Scrubbing is available only if your organization is enabled as an Early Adopter. To enable this option, navigate to your organization's settings and enable the "Early Adopter" option. Turning on this option allows access to features prior to full release, and can be disabled at any time.

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+
+You will then get access to one new option in both organization settings and the settings of each project called "Advanced datascrubber configuration". Its content is a piece of JSON-based configuration in the format described in the next section.
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "Early adopters have access to a new option ("Advanced datascrubber configuration") in both organization settings as well as the setting of each project, which creates a JSON-based configuration as described in the following section."

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 8]
```diff
@@ -0,0 +1,425 @@
...
+
+Note that everything you configure there will have direct impact on all new events, just as all the other data privacy-related settings do. However, it is not possible to break or undo any other data privacy settings that you may have configured. In other words, it is only possible to accidentally remove too much data, not too little.
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Nice!

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 9]
```diff
@@ -0,0 +1,425 @@
...
+
+If you have any questions related to this feature, feel free to contact us at `markus@sentry.io`.
+
+## A basic example
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Initial caps for titles (A Basic Example)

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 10]
```diff
@@ -0,0 +1,425 @@
...
+
+Say you have an exception message which, unfortunately, contains IP addresses which are not supposed to be there. You'd write:
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "One example is an exception message that includes an IP address that needs to be removed. In this example, you'd write:"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 11]
```diff
@@ -0,0 +1,425 @@
...
+    "$string": ["@ip:replace"]
+  }
+}
+```
+
+It reads as "replace all IP addresses in all strings", or "apply `@ip:replace` to all `$string` fields".
+
+`@ip:replace` is called a rule, and `$string` is a [_selector_](#selectors).
+
+## Built-in rules
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Init caps, please, so "Built-In Rules"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 12]
```diff
@@ -0,0 +1,425 @@
...
+
+The following rules exist by default:
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: "exist" is a bit unclear, do we mean that the following rules are provided as defaults for a user to adopt?

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 13]
```diff
@@ -0,0 +1,425 @@
...
+- `@ip:replace` and `@ip:hash` for replacing IP addresses.
+- `@imei:replace` and `@imei:hash` for replacing IMEIs
+- `@mac:replace`, `@mac:mask` and `@mac:hash` for matching MAC addresses
+- `@email:mask`, `@email:replace` and `@email:hash` for matching email addresses
+- `@creditcard:mask`, `@creditcard:replace` and `@creditcard:hash` for matching creditcard numbers
+- `@userpath:replace` and `@userpath:hash` for matching local paths (e.g. `C:/Users/foo/`)
+- `@password:remove` for removing passwords. In this case we're pattern matching against the field's key, whether it contains `password`, `credentials` or similar strings.
+- `@anything:remove`, `@anything:replace` and `@anything:hash` for removing, replacing or hashing any value. It is essentially equivalent to a wildcard-regex, but it will also match much more than strings.
+
+## Writing your own rules
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: "Writing Your Own Rules"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 14]
```diff
@@ -0,0 +1,425 @@
...
+Rules generally consist of two parts:
+
+- *Rule types* describe what to match. See [_Rule Types_](#rule-types) for an exhaustive list.
+- *Rule redaction methods* describe what to do with the match. See [_Redaction Methods_](#redaction-methods) for a list.
+
+Each page comes with examples. Try those examples out by pasting them into the "PII config" column of [_Piinguin_](https://getsentry.github.io/piinguin) and clicking on fields to get suggestions.
+
+## Selectors
+
+You have the possibility to select a region of the event using JSON-path-like syntax. To delete a specific key in "Additional Data", you could for example use:
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "You can select a region of the event using JSON-path-like syntax. As an example, to delete a specific key in "Additional Data":

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 15]
```diff
@@ -0,0 +1,425 @@
...
+
+```json
+{
+  "applications": {
+    "extra.foo": ["@anything:remove"]
+  }
+}
+```
+
+### Boolean logic
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: "Boolean Logic"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 16]
```diff
@@ -0,0 +1,425 @@
...
+* Build the disjunction (OR) using `||`, such as: `foo || bar` to match `foo` or `bar`.
+
+### Wildcards
+
+* `**` matches all subpaths, so that `foo.**` matches all JSON keys within `foo`.
+* `*` matches a single path item, so that `foo.*` matches all JSON keys one level below `foo`.
+
+### Value types
+
+The following can be used to select subsections by JSON-type or semantic meaning.
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Helpful to remove passive voice where possible, suggest "Select subsections by JSON-type or semantic meaning using the following:"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 17]
```diff
@@ -0,0 +1,425 @@
...
+
+  ```json
+  {
+    "applications": {
+      "$frame.vars": ["@anything:remove"]
+    }
+  }
+  ```
+
+### Escaping specal characters
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Init caps "Escaping Special Characters"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 18]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    }
+  },
+  "applications": {
+    "$string": ["hash_imei"]
+  }
+}
+```
+
+### mac
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "MAC"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 19]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    }
+  },
+  "applications": {
+    "$string": ["hash_device_id"]
+  }
+}
+```
+
+### imei
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "IMEI"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 20]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    }
+  },
+  "applications": {
+    "$string": ["hash_mac"]
+  }
+}
+```
+
+### ip
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: suggest "IP"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 21]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    }
+  },
+  "applications": {
+    "$string": ["hash_ip"]
+  }
+}
+```
+
+### creditcard
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest we use the word "Credit Card" here and in line 229 (lower case in 229)

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 22]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    }
+  },
+  "applications": {
+    "$string": ["hash_cc"]
+  }
+}
+```
+
+### userpath
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "User Path"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 23]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    }
+  },
+  "applications": {
+    "$string": ["remove_everything"]
+  }
+}
+```
+
+### multiple
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Would this more clear if the title read "Combine Multiple Rules"?

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 24]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    }
+  },
+  "applications": {
+    "$string": ["remove_ips_and_macs"]
+  }
+}
+```
+
+### alias
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Suggest "Alias Rules"

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 25]
```diff
@@ -130,6 +130,10 @@ You can choose to expand the keys which are scrubbed by the server, as well as p
 
 As mentioned earlier, configure scrubbing within SDK if possible so that sensitive data is not sent with the request.
 
+### Advanced server-side data scrubbing
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/sensitive-data.md]: Initial caps for titles, so "Advanced Server-Side Data Scrubbing"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md [hunk 26]
```diff
@@ -0,0 +1,425 @@
...
+
+Alias one rule to the other. This is the same as `multiple` except that you can only wrap one rule.
+
+```javascript
```
Comment (PeloWriter) [inline src/collections/_documentation/data-management/advanced-datascrubbing.md]: Here, and the following heads, suggest use of initial caps for Titles

File: src/collections/_documentation/data-management/advanced-datascrubbing.md (hunk 1 above)
Comment (untitaker) [reply src/collections/_documentation/data-management/advanced-datascrubbing.md]: done

File: src/collections/_documentation/data-management/advanced-datascrubbing.md (hunk 2 above)
Comment (untitaker) [reply src/collections/_documentation/data-management/advanced-datascrubbing.md]: done

File: src/collections/_documentation/data-management/advanced-datascrubbing.md (hunk 4 above)
Comment (untitaker) [reply src/collections/_documentation/data-management/advanced-datascrubbing.md]: done

File: src/collections/_documentation/data-management/advanced-datascrubbing.md (hunk 12 above)
Comment (untitaker) [reply src/collections/_documentation/data-management/advanced-datascrubbing.md]: updated with "can be used by default"

File: src/collections/_documentation/data-management/advanced-datascrubbing.md (hunk 18 above)
Comment (untitaker) [reply src/collections/_documentation/data-management/advanced-datascrubbing.md]: Those were supposed to be the rule names themselves, like on https://docs.sentry.io/error-reporting/configuration/?platform=javascript. I wrapped them in code tags to make them look more like that page

File: src/collections/_documentation/data-management/advanced-datascrubbing.md (hunk 26 above)
Comment (untitaker) [reply src/collections/_documentation/data-management/advanced-datascrubbing.md]: `javascript` only tells you which syntax highlighting to use for the codeblock, it's not a title or a word visible to the reader

**Pull request merged**